akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -512,6 +512,13 @@ object Source {
   def future[T](futureElement: Future[T]): Source[T, NotUsed] =
     fromGraph(new FutureSource[T](futureElement))
 
+  /**
+   * Never emits any elements, never completes and never fails.
+   * This stream could be useful in tests.
+   */
+  def never[T]: Source[T, NotUsed] =
+    future(Future.never)
```

Comment:
Add a name attribute in `DefaultAttributes.never` (see the other name attributes there for examples) and `.withAttributes(DefaultAttributes.never))` here, it is good for debuggability.

The Source could potentially be a single cached instance but I don't think this will be used much outside of tests so probably ok with a new identical instance each time it is used like it is.